acbuild/acbuild.go

```diff
@@ -156,6 +163,41 @@ func stdout(format string, a ...interface{}) {
 	fmt.Fprintln(os.Stdout, strings.TrimSuffix(out, "\n"))
 }
 
+func getLock() error {
+	err := os.MkdirAll(path.Join(contextpath, workprefix), 0755)
+	if err != nil {
+		return err
+	}
+
+	lockfile, err = os.OpenFile(lockpath(), os.O_CREATE|os.O_RDWR, 0644)
+	if err != nil {
+		return err
+	}
+
+	err = syscall.Flock(int(lockfile.Fd()), syscall.LOCK_EX)
```

Comment:
You're going to want LOCK_NB here, and probably to propagate a nicer error in the case that it's already locked.